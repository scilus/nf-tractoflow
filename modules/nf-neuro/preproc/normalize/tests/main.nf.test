nextflow_process {

    name "Test Process PREPROC_NORMALIZE"
    script "../main.nf"
    process "PREPROC_NORMALIZE"

    tag "modules"
    tag "modules_nfcore"
    tag "preproc"
    tag "preproc/normalize"

    tag "subworkflows"
    tag "subworkflows/load_test_data"

    setup {
            run("LOAD_TEST_DATA", alias: "LOAD_DATA") {
                script "../../../../../subworkflows/nf-neuro/load_test_data/main.nf"
                process {
                    """
                    input[0] = Channel.from( [ "heavy.zip" ] )
                    input[1] = "test.load-test-data"
                    """
                }
            }
        }

    test("preproc - normalize") {
        config "./nextflow.config"
        when {
            process {
                """
                input[0] = LOAD_DATA.out.test_data_directory
                    .map{ test_data_directory -> [
                    [ id:'test', single_end:false ], // meta map
                    file("\${test_data_directory}/dwi/dwi.nii.gz", checkIfExists: true),
                    file("\${test_data_directory}/dwi/dwi.bval", checkIfExists: true),
                    file("\${test_data_directory}/dwi/dwi.bvec", checkIfExists: true),
                    file("\${test_data_directory}/dwi/mask.nii.gz", checkIfExists: true)
                    ]}
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }
    }

    test("preproc - normalize - dti shells") {
        config "./nextflow_with_dti_shells.config"
        when {
            process {
                """
                input[0] = LOAD_DATA.out.test_data_directory
                    .map{ test_data_directory -> [
                    [ id:'test', single_end:false ], // meta map
                    file("\${test_data_directory}/dwi/dwi.nii.gz", checkIfExists: true),
                    file("\${test_data_directory}/dwi/dwi.bval", checkIfExists: true),
                    file("\${test_data_directory}/dwi/dwi.bvec", checkIfExists: true),
                    file("\${test_data_directory}/dwi/mask.nii.gz", checkIfExists: true)
                    ]}
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }
    }
}
