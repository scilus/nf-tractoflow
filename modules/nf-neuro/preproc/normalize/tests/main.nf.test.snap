{
    "preproc - normalize - dti shells": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test__dwi_normalized.nii.gz:md5,0d1ec91a9fdac4448bd7690fbf93a960"
                    ]
                ],
                "1": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test_fa_wm_mask.nii.gz:md5,9056862830ea449d4c4cf01752b51944"
                    ]
                ],
                "2": [
                    "versions.yml:md5,de4076b60c7dbf985efd96d5e6dc3faf"
                ],
                "dwi": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test__dwi_normalized.nii.gz:md5,0d1ec91a9fdac4448bd7690fbf93a960"
                    ]
                ],
                "mask": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test_fa_wm_mask.nii.gz:md5,9056862830ea449d4c4cf01752b51944"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,de4076b60c7dbf985efd96d5e6dc3faf"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-05-12T02:10:28.526169"
    },
    "preproc - normalize": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test__dwi_normalized.nii.gz:md5,0644f1faa8ecc7890747439800f7a616"
                    ]
                ],
                "1": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test_fa_wm_mask.nii.gz:md5,2b5c00d69e49c77bfec11f4cc1da2b8e"
                    ]
                ],
                "2": [
                    "versions.yml:md5,de4076b60c7dbf985efd96d5e6dc3faf"
                ],
                "dwi": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test__dwi_normalized.nii.gz:md5,0644f1faa8ecc7890747439800f7a616"
                    ]
                ],
                "mask": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test_fa_wm_mask.nii.gz:md5,2b5c00d69e49c77bfec11f4cc1da2b8e"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,de4076b60c7dbf985efd96d5e6dc3faf"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-05-12T02:10:08.116591"
    }
}